src/win/pipe.c

```diff
@@ -2299,6 +2299,22 @@ int uv_pipe_getpeername(const uv_pipe_t* handle, char* buffer, size_t* size) {
 }
 
 
+int uv_pipe_get_credentials(const uv_pipe_t* handle, uv_credentials_t *creds) {
+  BOOL r;
+  ULONG pid;
+
+  r = GetNamedPipeClientProcessId(handle->handle, &pid);
```

Comment:
How would I know which is which? I don't see a function that discerns between the two.

Reply:
Also I'd be skeptical of this logic because we're mostly looking at the client for credentials. The goal of the function is mostly to check if connecting clients are allowed to connect and interact with the server.

Reply:
A client may want to validate it is not connecting to a rouge server so I think it can still be useful. 

As far as determining if the pipe is a server or client, it looks like the `GetNamedPipeInfo()` Win32 API can tell you based on the presence of the `PIPE_SERVER_END` flag... But it's odd that there isn't a single GetNamedPipePeerProcessId() API or such that would do the right thing regardless of whether it is a server or a client...

Reply:
Possible simple test: if the result is our process pid, then we should return the pid of the other end (even if that happens to also be ourself)

Reply:
@vtjnash that feels extremely hack-ish. :/

Reply:
I know that you could probably also use GetNamedPipeInfo, but my method seemed to also be equally precise, if a bit unexpected.